src/github/push.unit.test.ts

```diff
@@ -0,0 +1,71 @@
+import { pushWebhookHandler } from "./push";
+import { WebhookContext } from "routes/github/webhook/webhook-context";
+import { getLogger } from "config/logger";
+import { GitHubPushData } from "../interfaces/github";
+import { enqueuePush } from "../transforms/push";
+
+jest.mock("../transforms/push");
+
+const GHES_GITHUB_INSTALLATION_ID = 1234;
+const GHES_GITHUB_APP_ID = 111;
+const GHES_GITHUB_UUID = "xxx-xxx-xxx-xxx";
+const GHES_GITHUB_APP_APP_ID = 1;
+const GHES_GITHUB_APP_CLIENT_ID = "client-id";
+
+describe("PushWebhookHandler", ()=>{
+	let jiraClient: any;
+	let util: any;
+	beforeEach(() => {
+		jiraClient = { baseURL: jiraHost };
+		util = null;
+	});
+	describe("GitHub Cloud", ()=>{
+		it("should NOT push GitHubAppConfig", async ()=>{
+			await pushWebhookHandler(getWebhookContext({ cloud: true }), jiraClient, util, GHES_GITHUB_INSTALLATION_ID);
+			expect(enqueuePush).toBeCalledWith(expect.anything(), expect.anything(), undefined);
+		});
+	});
+	describe("GitHub Enterprise Serer", ()=>{
+		it("should push GitHubAppConfig", async ()=>{
+			await pushWebhookHandler(getWebhookContext({ cloud: false }), jiraClient, util, GHES_GITHUB_INSTALLATION_ID);
+			expect(enqueuePush).toBeCalledWith(expect.anything(), expect.anything(), expect.objectContaining({
+				uuid: GHES_GITHUB_UUID,
+				gitHubAppId: GHES_GITHUB_APP_ID,
+				appId: GHES_GITHUB_APP_APP_ID,
+				clientId: GHES_GITHUB_APP_CLIENT_ID,
+				gitHubBaseUrl: gheUrl
+			}));
+		});
+	});
+	const getWebhookContext = ({ cloud }: {cloud: boolean}) => {
+		const payload: GitHubPushData = {
+			installation: {
+				id: GHES_GITHUB_INSTALLATION_ID,
+				node_id: 123
+			},
+			webhookId: "aaa-bbb-ccc",
+			webhookReceived: Date.now(),
+			repository: { } as GitHubPushData["repository"], //force it as not required in test
+			commits: [{
+				id: "commit-1",
+				message: "ARC-0001 some commit message"
+			}]
+		};
+		return new WebhookContext({
+			id: "1",
+			name: "push",
+			log: getLogger("test"),
+			payload,
+			...(cloud ? undefined : {
+				gitHubAppConfig: {
+					uuid: GHES_GITHUB_UUID,
+					gitHubAppId: GHES_GITHUB_APP_ID,
+					appId: GHES_GITHUB_APP_APP_ID,
+					clientId: GHES_GITHUB_APP_CLIENT_ID,
+					gitHubBaseUrl: gheUrl
+				}
```

Comment:
why does cloud not have the githubAppConfig?  It's not GHE specific.  Also, no need to use deconstructor, just `gitubAppConfig: cloud ? undefined : { config things }`

Reply:
`why does cloud not have the githubAppConfig` <--
1. Seems many places in the code, we have this impression, `if cloud? then undefined, else if GHE, then some concrete value`. Don't want to break that yet.
2. I suspect majority traffic of the webhook event is cloud. Adding cloud appId(and others) to every single queue msg will big impact. Not like GHE we have to put at least one id in the payload, for cloud we can get away without any id. So given the impact is big, I am hesitate to put it in the payload now. 

--------------------

Can we think about it, and if decided to put cloud, do it in another PR?

-------------

For `just gitubAppConfig: cloud ? undefined : { config things }`, it is me always been thinking there's a different of an object without key `{}` vs and `object with a key, but value is undefined`. But on a second thoughts, those seems no differences from coding point of view. So I revert the typing to `gitHubAppConfig?: GitHubAppConfig` from `gitHubAppConfig: GitHubAppConfig | undefined`

Reply:
1. Put cloud in
2. Check api.github vs github

Reply:
fixed.